src/main/scala/org/scalajs/tools/tsimporter/parser/TSDefParser.scala

```diff
@@ -105,7 +117,20 @@ class TSDefParser extends StdTokenParsers with ImplicitConversions {
     "enum" ~> typeName ~ ("{" ~> ambientEnumBody <~ "}") ^^ EnumDecl
 
   lazy val ambientEnumBody: Parser[List[Ident]] =
-    repsep(identifier <~ opt("=" ~ (numericLit | stringLit) ), ",") <~ opt(",")
+    repsep(identifier <~ opt("=" ~! valueExpression), ",") <~ opt(",")
```

Comment:
What are the `...!` operators?

Reply:
they are non backtracking operators. basically, after the `=` is parsed here the parser will not go back and try other alternatives prior to `=` when an error comes afterwards. this is very important to get better error locations when parsing fails, because we know that when we arrived after the `=` there is no other rule that could parse this. so there is no point in going back and try to parse a class declaration instead of the enum declaration, for example. Instead we just stop and show the much more accurate error location.

I was already about to make a pr which added many more of these operators but it's not always totally clear where to introduce these cuts while preserving the semantics and I want to make sure that I am correct. I will make the pr in a few days